src/net.cpp

```diff
@@ -2331,6 +2332,11 @@ bool CAddrDB::Read(CAddrMan& addr)
 
     CDataStream ssPeers(vchData, SER_DISK, CLIENT_VERSION);
 
+    return Read(addr, ssPeers, hashIn);
+}
+
+bool CAddrDB::Read(CAddrMan& addr, CDataStream& ssPeers, uint256& hashIn)
+{
     // verify stored checksum matches input data
     uint256 hashTmp = Hash(ssPeers.begin(), ssPeers.end());
```

Comment:
>Depends on the (non-test) code impact. 

It is on my list of things to investigate, but I was considering creating a class which wraps File/fopen and can be mocked out in unitests similar to RandomInt in addrman.h.

https://github.com/bitcoin/bitcoin/blob/master/src/addrman.h#L239

I share your concern as this would require a big diff when all the instances of File are replaced with mockable File. I assume this is a problem others have solved so I'm looking into alternates.

>I'm not entirely sure that is necessary - e.g. you could create a temporary file and make it read that.

YMMV, but in my experience tmp files in unittests cause hard to debug test failures due to filesystem weirdness and added state (file deletes sometimes fail).